src/GenericHostBuilderExample/Program.cs

```diff
@@ -1,20 +1,45 @@
 ﻿using System.Threading.Tasks;
 using Autofac;
 using Autofac.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Hosting;
 
 namespace GenericHostBuilderExample
 {
     internal class Program
     {
-        private static async Task Main() =>
-            await new HostBuilder()
+        public static async Task Main(string[] args)
+        {
+            // The `UseServiceProviderFactory(new AutofacServiceProviderFactory())` call here allows for
+            // ConfigureContainer to be supported with
+            // a strongly-typed ContainerBuilder. If you don't
+            // have the call to `UseServiceProviderFactory(new AutofacServiceProviderFactory())` here, you won't get
+            // ConfigureContainer support. This also automatically
+            // calls Populate to put services you register during
+            // ConfigureServices into Autofac.
+            await Host.CreateDefaultBuilder(args)
                 .UseServiceProviderFactory(new AutofacServiceProviderFactory())
-                .ConfigureContainer<ContainerBuilder>((context, builder) =>
-                {
-                    builder.RegisterType<Logger>().As<ILogger>();
-                    builder.RegisterType<HostedService>().As<IHostedService>();
-                })
+                .ConfigureServices(ConfigureServices)
+                .ConfigureContainer<ContainerBuilder>(ConfigureContainer)
                 .RunConsoleAsync();
+        }
+
+
+        private static void ConfigureServices(IServiceCollection services)
+            // Use extensions from libraries to register services in the
+            // collection. These will be automatically added to the
+            // Autofac container.
+            => services.AddHostedService<HostedService>();
+
+        private static void ConfigureContainer(ContainerBuilder containerBuilder)
+        {
+            // Add any Autofac modules or registrations.
+            // This is called AFTER ConfigureServices so things you
+            // register here OVERRIDE things registered in ConfigureServices.
+            //
+            // You must have the call to `UseServiceProviderFactory(new AutofacServiceProviderFactory())`
```

Comment:
Is this true? Or is it a copy/paste error from the ASP.NET example?

Reply:
You mean if `ConfigureContainer` is called after `ConfigureServices`? Yes, that is true.

Reply:
No, I meant the "you must have the UseSerivceProviderFactory" bit. Is that part true, too?

Reply:
Since we have removed it from the `Autofac.Extensions.DependencyInjection` I thought it would be the right thing to describe it in the new fashion. If you want I can change it and / or add another sample.

Reply:
Sorry, perhaps I'm miscommunicating. There are two statements in the comment:

1. `ConfigureContainer` is called after `ConfigureServices`.
2. You must have `UseServiceProviderFactory` in order for `ConfigureContainer` to be called at all.

I know that statement 1 is true. I was not clear that statement 2 was also true, hence I was asking for clarification. It makes _logical sense_ as I think more about it, however, since I've not tried it I didn't know what would happen.

Reply:
Ah, now I got it. Thanks for the clarification.

I just double checked because I wasn't sure. The hosted-service is not being activated without calling `UseServiceProviderFactory(new AutofacServiceProviderFactory())`. The host would just run without doing anything, not even throwing an error.

In any case what is kind of redundant is the `ConfigureContainer<ContainerBuilder>(ConfigureContainer)` call. It could be passed into the former call as well. Like so:
`UseServiceProviderFactory(new AutofacServiceProviderFactory(ConfigureContainer))`